modules/engines/engine-xsodata/src/main/java/com/sap/xsk/xsodata/ds/handler/XSKScriptingOData2EventHandler.java

```diff
@@ -0,0 +1,501 @@
+import org.eclipse.dirigible.engine.odata2.transformers.DBMetadataUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import javax.sql.DataSource;
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class XSKScriptingOData2EventHandler extends ScriptingOData2EventHandler {
+
+  private static final Logger logger = LoggerFactory.getLogger(XSKScriptingOData2EventHandler.class);
+  private static final String ERROR_WHEN_PREPARING_TEMPORARY_TABLE_SQL = "Error when preparing temporary table SQL: ";
+
+  private static final String ODATA2_EVENT_HANDLER_NAME = "xsk-odata-event-handler";
+
+  private DataSource dataSource = (DataSource) StaticObjects.get(StaticObjects.DATASOURCE);
+
+  private static final String DUMMY_BUILDER = "dummyBuilder";
+  private static final String SELECT_BUILDER = "selectBuilder";
+  private static final String INSERT_BUILDER = "insertBuilder";
+  private static final String UPDATE_BUILDER = "updateBuilder";
+  private static final String SQL_CONTEXT = "sqlContext";
+
+  private static final String CONNECTION = "connection";
+  private static final String BEFORE_TABLE_NAME = "beforeTableName";
+  private static final String AFTER_TABLE_NAME = "afterTableName";
+  private static final String ON_CREATE_ENTITY_TABLE_NAME = "onCreateEntityTableName";
+  private static final String BEFORE_UPDATE_ENTITY_TABLE_NAME = "beforeUpdateEntityTableName";
+  private static final String BEFORE_DELETE_ENTITY_TABLE_NAME = "beforeDeleteEntityTableName";
+  private static final String ENTRY_MAP = "entryMap";
+
+  @Override
+  public void beforeCreateEntity(PostUriInfo uriInfo, String requestContentType, String contentType, ODataEntry entry,
+      Map<Object, Object> context) {
+    SQLInsertBuilder dummyBuilder = (SQLInsertBuilder) context.get(DUMMY_BUILDER);
+    SQLInsertBuilder insertBuilder = (SQLInsertBuilder) context.get(INSERT_BUILDER);
+    SQLContext sqlContext = (SQLContext) context.get(SQL_CONTEXT);
+
+    Connection connectionParam = null;
+    try {
+      connectionParam = dataSource.getConnection();
+
+      String targetTableName = getSQLInsertBuilderTargetTable(dummyBuilder, sqlContext);
+      String afterTableName = generateTemporaryTableName(uriInfo.getTargetType().getName());
+
+      createTemporaryTableLikeTable(connectionParam, afterTableName, targetTableName);
+      insertIntoTemporaryTable(connectionParam, insertBuilder, afterTableName, sqlContext);
+
+      context.put(CONNECTION, connectionParam);
+      context.put(AFTER_TABLE_NAME, afterTableName);
+
+      super.beforeCreateEntity(uriInfo, requestContentType, contentType, entry, context);
+    } catch (Exception e) {
+      logger.error(e.getMessage(), e);
+      throw new InvalidStateException(e);
```

Comment:
just re-throw, it will be better in this case to create specific custom RuntimeException, something like EventHandlerException